source/fundamentals/connection.txt

```diff
@@ -4,16 +4,201 @@ Connection Guide
+store the same set of data. This configuration provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname and port numbers
+of each instance, separated by commas, and the replica set name as the value
+of the ``replicaSet`` parameter in the connection string. In the following
+example, the hostnames are ``host1``, ``host2``, and ``host3``, and the
+port numbers are all ``27017``. The replica set name is ``myRS``.
+
+.. code-block:: none
+
+   mongodb://host1:27017,host2:27017,host3:27017/?replicaSet=myRS
+
+When connecting to a replica set, the driver takes the following actions by default:
+
+- Discovers all replica set members when given the address of any one member.
+- Dispatches operations to the appropriate member, such as instructions
+  to write against the **primary**.
+
+.. tip::
+
+   You only need to specify one host to connect to a replica set.
+   However, to ensure connectivity when the specified host
+   is unavailable, you should provide the full list of hosts.
+
+Direct Connection
++++++++++++++++++
+
+To force operations on the host designated in the connection URI, specify the ``directConnection`` option. Direct
+connections:
+
+- Don't support SRV strings.
+- Fail on writes when the specified host is not the **primary**.
+- Require you to :manual:`enable secondary reads </reference/method/rs.secondaryOk/>`
+  when the specified host isn't the **primary**.
+
+.. _connection-options:
+
+------------------
+Connection Options
+------------------
+
+This section explains several common MongoDB connection and authentication
+options. You can pass the connection options as parameters of the connection
+URI to specify the behavior of the client. 
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 10 15 40
+
+   * - Option Name
+     - Type
+     - Default Value
+     - Description
+
+   * - **connectTimeoutMS**
+     - integer
+     - ``30000``
+     - Specifies the number of milliseconds to wait before timeout on a TCP
+       connection.
+
+   * - **maxPoolSize**
+     - integer
+     - ``100``
+     - Specifies the maximum number of connections that a connection pool may
+       have at a given time.
+
+   * - **maxIdleTimeMS**
+     - integer
+     - ``0``
+     - Specifies the maximum amount of time a connection can remain idle
+       in the connection pool before being removed and closed.
+
+   * - **minPoolSize**
+     - integer
+     - ``0``
+     - Specifies the minimum number of connections that must exist at any moment
+       in a single connection pool.
+
+   * - **socketTimeoutMS**
+     - integer
+     - ``0``
+     - Specifies the number of milliseconds to wait to find an
+       available, suitable server to execute an operation.
+
+   * - **serverSelectionTimeoutMS**
+     - integer
+     - ``30000``
+     - Specifies the number of milliseconds to wait before timeout on a TCP
+       socket.
+
+   * - **heartbeatIntervalMS**
```

Comment:
```suggestion
   * - **heartbeatFrequencyMS**
```

The Go driver supports both `heartbeatIntervalMS` and `heartbeatFrequencyMS`. `heartbeatFrequencyMS` is preferred since it is portable between drivers. It is specified in the [URI options specification](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst#list-of-specified-options).